sql/parser/builtins.go

```diff
@@ -761,3 +773,73 @@ func datumToRawString(datum Datum) (string, error) {
 
 	return "", fmt.Errorf("argument type unsupported: %s", datum.Type())
 }
+
+// Round rounds x to n digits.
+// Directly taken from CPython's `double_round` in
+// cpython/Object/floatobject.c.
+//
+// TODO(thschroeter):
+// - understand how it works
```

Comment:
Short answer for how this works: Multiply `x` by `pow(10, n)` to shift the position of the decimal point, round to the nearest even int, then divide by `pow(10, n)` to shift the decimal point back to its original position. As far as I can tell, this should work for negative `n`. See http://play.golang.org/p/0JxP---AGU.